R/g.part4_extractid.R

```diff
@@ -30,57 +30,34 @@ g.part4_extractid = function(idloc, fname, dolog, sleeplogidnum, sleeplog, accid
       accid = newaccid[1]
     }
   }
-  if (sleeplogidnum == TRUE) {
-    # remove last character (in some studies numeric id is followed by character)
-    accid_bu = accid
-    getLastCharacterValue = function(x) {
-      tmp = as.character(unlist(strsplit(x,"")))
-      return(tmp[length(tmp)])
-    }
-    letter = apply(as.matrix(accid), MARGIN = c(1), FUN = getLastCharacterValue)
-    for (h in 1:length(accid)) {
-      options(warn = -1)
-      numletter = as.numeric(letter[h])
-      options(warn = 0)
-      if (is.na(numletter) == TRUE) { # do not remove latest character if it is a number
-        accid[h] = as.character(unlist(strsplit(accid[h],letter[h]))[1])
-      }
-    }
-    accid = suppressWarnings(as.numeric(accid))
-    #catch for files with only id in filename and for whom the above attempt to extract the id failed:
-    if (is.na(accid) == TRUE) accid = accid_bu
-  }
   # get matching identifier from sleeplog
-  if (dolog == TRUE) {
-    accid_num = suppressWarnings(as.numeric(accid))
-    if (sleeplogidnum == FALSE) {
-      # remove spaces in ID, to ease matching, because some accelerometer brands at several spaces behind ID
-      sleeplog$ID = gsub(pattern = " ", replacement = "", x = as.character(sleeplog$ID))
-      accid = gsub(pattern = " ", replacement = "", x = as.character(accid))
-      # attempt to match
-      matching_indices_sleeplog = which(as.character(sleeplog$ID) == as.character(accid))
-      if (length(matching_indices_sleeplog) == 0) {
-        matching_indices_sleeplog_alternative = which(sleeplog$ID == accid_num)
-        if (length(matching_indices_sleeplog_alternative) > 0) {
-          warning("\nArgument sleeplogidnum is set to FALSE, but it seems the identifiers are
-                    stored as numeric values, you may want to consider changing sleeplogidnum to TRUE")
-        }
+  if (dolog == TRUE) { # since now we do not have sleeplogidnum, we only need to check the last letter of accid, if sleeplog is available
+    if (suppressWarnings(!is.na(as.numeric(sleeplog$ID[1]))) & # sleeplog id is numeric 
```

Comment:
Why did you add this `is.na(as.numeric` check?
Now all IDs are interpreted as character this should not be needed anymore, right?

It seems we need a new argument to specify how many characters from the end of the ID should be ignored, because it is now applied to all IDs that end with a character, which may not be desirable. Another option is to have an argument to strip all letters from the ID, which may be more generally useful.

Reply:
I think it would be better not to deal with fixing IDs within GGIR and instruct users to make sure their IDs are coherent between logs and accelerometer files. I would just return a warning when IDs do not match to call the user's attention. What do you think?

Otherwise, I am not sure if many people rely on this functionality of removing the last letter. If so, should we add the argument (e.g.) `logID_ignoreLetters`? (log instead of sleeplog because this could also be used for the activity log and the data cleaning file)

Reply:
Update: I think the latest commits resolve this without the need of a new argument.